src/arch/xen/boot.S

```diff
@@ -1,42 +1,40 @@
 /**
  * @file
- * @brief bootstrap the kernel
+ * @brief bootstrap the Xen kernel
  *
  * @date 10.11.10
  * @author Nikolay Korotky
+ * @author Andrey Golikov
+ * @author Anton Kozlov
  */
 
-
-#include <asm/linkage.h>
 #include <hal/cpu.h>
 
-	.section .text
-	.align 4
+#define C_ENTRY(name) .globl name; .align 2; name
```

Comment:
Do you really want to align .text at 4 but function's name at 2?
I think this definitions should be extracted to a header